### PR TITLE
Prevent search on discard of empty chip

### PR DESCRIPTION
**Pull request by rsasch**
see https://elastc.com/c/46LPfbYI/291-don-t-trigger-search-on-discard-of-empty-chip

### CODE REVIEW

File: ui/src/app/shared/header/header.component.ts [hunk 1]
```diff
@@ -174,8 +174,13 @@ export class HeaderComponent implements OnInit, AfterViewInit, AfterViewChecked,
...
 
   removeChip(chipKey: string): void {
-    this.chips.delete(chipKey);
-    this.search();
+    if (this.chips.get(chipKey).valueOf()) {
+      this.chips.delete(chipKey);
+      this.search();
+    }
+    else {
+      this.chips.delete(chipKey);
```
Comment (bfcrampton) [inline ui/src/app/shared/header/header.component.ts]: nit: move out of the else:
```
this.chips.delete(chipKey);
if (this.chips.get(chipKey).valueOf()) {
  this.search();
}
```

Comment (rsasch) [reply ui/src/app/shared/header/header.component.ts]: If the chip has been deleted, `this.chips.get(chipKey)` returns an error; I can assign it to a variable and then use that in the if statement, if you think that would be cleaner.

File: ui/src/app/shared/header/header.component.ts [hunk 2]
```diff
@@ -174,8 +174,13 @@ export class HeaderComponent implements OnInit, AfterViewInit, AfterViewChecked,
   }
 
   removeChip(chipKey: string): void {
-    this.chips.delete(chipKey);
-    this.search();
+    if (this.chips.get(chipKey).valueOf()) {
+      this.chips.delete(chipKey);
+      this.search();
+    }
```
Comment (calbach) [inline ui/src/app/shared/header/header.component.ts]: `} else {`

The rest of the codebase uses this convention.

File: ui/src/app/shared/header/header.component.ts [hunk 3]
```diff
@@ -174,8 +174,13 @@ export class HeaderComponent implements OnInit, AfterViewInit, AfterViewChecked,
   }
 
   removeChip(chipKey: string): void {
-    this.chips.delete(chipKey);
-    this.search();
+    if (this.chips.get(chipKey).valueOf()) {
+      this.chips.delete(chipKey);
+      this.search();
```
Comment (calbach) [inline ui/src/app/shared/header/header.component.ts]: Per the other ticket about "?q=", this special-casing shouldn't be necessary if we omit q= on an unfiltered query. I would just fix that other issue and leave this part of the code unchanged.

**Pull request closed without merging**